torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2120,6 +2184,36 @@ def _make_tensor_helper(shape, low=None, high=None):
                    dtypesIfCPU=all_types_and_complex_and(torch.half, torch.bfloat16),
                    dtypesIfCUDA=all_types_and_complex_and(torch.half, torch.bfloat16),
                    assert_autodiffed=True,),
+    OpInfo('pow',
+           op=torch.pow,
+           dtypes=all_types_and_complex_and(torch.half, torch.bfloat16),
+           dtypesIfCUDA=all_types_and_complex_and(torch.half, torch.bfloat16),
+           dtypesIfCPU=all_types_and_complex_and(torch.half, torch.bfloat16),
+           sample_inputs_func=sample_inputs_pow,
+           test_inplace_grad=False,
+           assert_autodiffed=True,
+           supports_out=True,
+           decorators=(precisionOverride({torch.float16: 1e-1, torch.bfloat16: 1e-1,
+                                          torch.float32: 1e-5, torch.float64: 1e-5}),),
+           skips=(
+               # Inplace variants currently use the same sample inputs as the method for which
+               # they are a variant, causing this test to fail because of broadcasting semantics.
+               # This test should be enabled after GitHub PR 53014 would land.
+               # Reference: https://github.com/pytorch/pytorch/issues/50747
+               SkipInfo('TestCommon', 'test_variant_consistency_eager',
+                        dtypes=[torch.bfloat16, torch.float16, torch.float32, torch.float64]),
+               # Due to AVX2 curently not being fully supported for Float16, log_vml_cpu can't be enabled
+               # for Float16, causing this test to fail.
+               SkipInfo('TestCommon', 'test_variant_consistency_jit',
+                        device_type='cpu', dtypes=[torch.float16]),
+               # JIT doesn't currently support complex literals.
+               # This test should be enabled after GitHub PR 52881 would land.
+               # Reference: https://github.com/pytorch/pytorch/issues/51996#issuecomment-791517296
+               SkipInfo('TestCommon', 'test_variant_consistency_jit',
+                        dtypes=[torch.complex64, torch.complex128]),
+               # Bool is currently supported in the sense that it's converted to other dtypes,
```

Comment:
Why don't we want to run it?

If we really think running the OpInfo tests with the bool dtype for pow is a bad idea then the correct way to skip them would be to update this field:

https://github.com/pytorch/pytorch/blob/1772e26f6380d189ba4e03da492ff18eaa139563/torch/testing/_internal/common_methods_invocations.py#L142

That lets the OpInfo tests know which dtypes they should run in.

Reply:
I haven't added `bool` to the `dtypes` list of `pow`'s `OpInfo`, as I think `pow` shouldn't be supported for it (please confirm as to whether that's indeed the case). But since `bool` is typecasted to another dtype if `pow` is invoked on it, `test_unsupported_types` fails for it. That's why I skipped this test for `bool`.

Reply:
Thank you for explaining that, @imaginary-person, I think I understand the issue now.

What happens if you replace the calls to rand() in the pow sample inputs with calls to make_tensor(), which we prefer using in the test suite? rand() does not support the bool dtype, but make_tensor() was carefully written to support all dtypes. make_tensor() should also allow you to get rid of _uniform_scalar(), since make_tensor() supports passing a range that the generated values should be in.

Reply:
Thank you for your advice, @mruberry!
I'm still a little confused, though - are there any use-cases for supporting `pow` for the `bool` dtype?

If I replace the calls to `rand()` with `make_tensor()`, then JIT & fn_grad, fn_grad_grad  tests fail due to asserts:

>   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_jit.py", line 108, in check_against_reference
>     self.assertTrue(torch.allclose(g2, g2_test, atol=5e-4, rtol=1e-4))
> AssertionError: False is not true

So, it seems I would have to specify a custom range in `make_tensor`.

Reply:
Specifying a custom range using arguments for `low`, `high` in `make_tensor` would work, so long as I produce tensors that only have positive numbers, as `pow` of negative floats produces NaNs if the fractional portion of the exponent isn't 0.

Reply:
`OpInfo` tests shouldn't be enabled for `bool`, as some tests would fail with Runtime errors pertaining to `pow` not supporting `bool`.

However, just like numpy, `bool` tensors in PyTorch are typecasted to other dtypes before dispatching a `pow` kernel, so `test_unsupported_types` fails for it, as it expects a Runtime error to be thrown, which doesn't actually happen. That's why it'd have to be skipped, although I didn't even add `bool` to the `OpInfo` `dtypes` list..

Reply:
> What happens if you replace the calls to rand() in the pow sample inputs with calls to make_tensor(), which we prefer using in the test suite?

Done. Used custom `low`, `high` arguments in calls to `make_tensor`.